### PR TITLE
Update kmscon, change upstream

### PR DESCRIPTION
**Pull request by RunningDroid**
#### Testing the changes
- I tested the changes in this PR: **YES**

### CODE REVIEW

File: srcpkgs/kmscon/template [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
  libpciaccess-devel eudev-libudev-devel dbus-devel fuse-devel"
 depends="xkeyboard-config"
 short_desc="Terminal emulator based on Kernel Mode Setting (KMS)"
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="MIT"
-homepage="http://www.freedesktop.org/wiki/Software/kmscon"
-distfiles="${FREEDESKTOP_SITE}/kmscon/releases/kmscon-${version}.tar.xz"
-checksum=0ab01c1cdc4fbd692ce765e80478bc2d9663a7c55a5c75cc7ac421366ee6ae2b
+homepage="https://github.com/Aetf/kmscon"
+distfiles="https://github.com/Aetf/kmscon/archive/refs/tags/v${version}.tar.gz"
```
Comment (kruceter) [inline srcpkgs/kmscon/template:15]:
```suggestion
distfiles="https://github.com/Aetf/kmscon/archive/v${version}.tar.gz"
```

File: srcpkgs/kmscon/template [hunk 2]
```diff
@@ -1,20 +1,21 @@
 # Template file for 'kmscon'
 pkgname=kmscon
-version=8
-revision=7
-build_style=gnu-configure
-configure_args="--disable-static --disable-multi-seat"
+version=9.0.0
+revision=1
+build_style=meson
+configure_args="-Dwerror=false"
```
Comment (kruceter) [inline srcpkgs/kmscon/template:6]:
```suggestion
configure_args="-Dwerror=false -Dtests=false"
```

`make_check` is set to `no`, but tests still build.

File: srcpkgs/libtsm/template [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
 makedepends="libxkbcommon-devel"
 short_desc="Terminal-emulator State Machine"
 maintainer="Orphaned <orphan@voidlinux.org>"
-license="MIT"
-homepage="http://www.freedesktop.org/wiki/Software/kmscon/libtsm/"
-distfiles="${FREEDESKTOP_SITE}/kmscon/releases/libtsm-${version}.tar.xz"
-checksum=114115d84a2bc1802683871ea2d70a16ddeec8d2f8cde89ebd2046d775e6cf07
+license="MIT, LGPL-2.1-or-later, BSD-2-Clause"
+homepage="https://github.com/Aetf/libtsm"
+distfiles="https://github.com/Aetf/libtsm/archive/refs/tags/v${version}.tar.gz"
```
Comment (kruceter) [inline srcpkgs/libtsm/template:13]:
```suggestion
distfiles="https://github.com/Aetf/libtsm/archive/v${version}.tar.gz"
```

File: srcpkgs/libtsm/template [hunk 4]
```diff
@@ -1,16 +1,17 @@
 # Template file for 'libtsm'
 pkgname=libtsm
-version=3
-revision=3
-build_style=gnu-configure
+version=4.0.2
+revision=1
+build_style=cmake
+configure_args="-DCMAKE_BUILD_TYPE=Release"
```
Comment (kruceter) [inline srcpkgs/libtsm/template:6]: Rebase and apply this patch:

[0001-fixup-libtsm-update-to-4.0.2-switch-to-Aetf-s-fork.txt](https://github.com/void-linux/void-packages/files/10326860/0001-fixup-libtsm-update-to-4.0.2-switch-to-Aetf-s-fork.txt)

File: srcpkgs/kmscon/template (hunk 1 above)
Comment (paper42) [reply srcpkgs/kmscon/template:15]: Please use refs/tags, just archive/ can return an error when there is a branch of the same name as tag

File: srcpkgs/libtsm/template (hunk 3 above)
Comment (paper42) [reply srcpkgs/libtsm/template:13]: -||-

**Review by kruceter — APPROVED** [review]
Alpine and Gentoo use Aetf's fork, too.

I have found no problems using it on x86_64-musl (at least without surprises after a couple of days of active usage thus far).

**Comment by Vaelatern** [comment]
Ok.

I don't like changing upstreams, but after reading the original dev email list (last post in 2017 pointing to a different fork) I think this is a fine change.

**Pull request merged**